modules/samples/producers/jms/src/main/java/org/wso2/carbon/sample/jmsclient/JMSClient.java

```diff
@@ -49,66 +51,130 @@ public static void main(String[] args) {
 
-                List<String> messagesList = JMSClientUtil.readFile(filePath);
-                try {
-                    if(format.equalsIgnoreCase("csv")){
-                        log.info("Sending Map messages on '" + topicName + "' topic");
-                        JMSClientUtil.publishMapMessage(producer, session, messagesList);
+                if(broker.equalsIgnoreCase("activemq")){
+                    properties.load(ClassLoader.getSystemClassLoader().getResourceAsStream("activemq.properties"));
+                    Context context = new InitialContext(properties);
+                    connFactory = (QueueConnectionFactory) context.lookup("ConnectionFactory");
 
-                    }else{
-                        log.info("Sending  " + format + " messages on '" + topicName + "' topic");
-                        JMSClientUtil.publishTextMessage(producer, session, messagesList);
+                }else if(broker.equalsIgnoreCase("mb")){
+                    properties.load(ClassLoader.getSystemClassLoader().getResourceAsStream("mb.properties"));
+                    Context context = new InitialContext(properties);
+                    connFactory = (QueueConnectionFactory) context.lookup("qpidConnectionFactory");
+
+                }else if(broker.equalsIgnoreCase("qpid")){
+                    properties.load(ClassLoader.getSystemClassLoader().getResourceAsStream("qpid.properties"));
+                    Context context = new InitialContext(properties);
+                    connFactory = (QueueConnectionFactory) context.lookup("qpidConnectionFactory");
+
+                }else{
+                    log.error("Please enter a valid JMS message broker. (ex: activemq, mb, qpid");
+                }
+
+                if(connFactory != null) {
+                    queueConnection = connFactory.createQueueConnection();
+                    queueConnection.start();
+                    session = queueConnection.createQueueSession(false, Session.AUTO_ACKNOWLEDGE);
+
+                    if (session != null) {
+                        Queue queue = session.createQueue(queueName);
+                        MessageProducer producer = session.createProducer(queue);
+
+                        List<String> messagesList = JMSClientUtil.readFile(filePath);
+                        try {
+                            if (format.equalsIgnoreCase("csv")) {
+                                log.info("Sending Map messages on '" + queueName + "' queue");
+                                JMSClientUtil.publishMapMessage(producer, session, messagesList);
+
+                            } else {
+                                log.info("Sending  " + format + " messages on '" + queueName + "' queue");
+                                JMSClientUtil.publishTextMessage(producer, session, messagesList);
+                            }
+                        } catch (JMSException e) {
+                            log.error("Can not subscribe." + e.getMessage(), e);
```

Comment:
Thanks @prabathabey, fixed it in the pull request - https://github.com/wso2/product-cep/pull/166